File: web/taxi-app-web/src/app/models/driver-profile-response.ts
import { UserProfileResponse } from "./user-profile-response";

export enum VehicleType {
  "STANDARD",
  "LUXURY",
  "VAN"
}

export interface DriverProfileResponse extends UserProfileResponse {
  workingHoursToday: number;
  vehicle: {
    id: number;
    brand: string;
    model: string;
    plateNumber: string;
    seatNumber: number;
    babyFriendly: boolean;
    petFriendly: boolean;
    type: VehicleType;
  }
}

File: mobile/TaxiApp/app/src/main/java/model/DriverProfileResponse.java
package model;

public class DriverProfileResponse  extends  UserProfileResponse {
    private double workingHoursToday;
    private Vehicle vehicle;

    public double getWorkingHoursToday() { return workingHoursToday; }
    public Vehicle getVehicle() { return vehicle; }

    public void setWorkingHoursToday(double workingHoursToday) { this.workingHoursToday = workingHoursToday; }
    public void setVehicle(Vehicle vehicle) { this.vehicle = vehicle; }
}
